File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/listener/ScheduleChangeListener.java
package io.orkes.conductor.scheduler.listener;

import io.orkes.conductor.scheduler.model.WorkflowSchedule;

/** Listener for changes to workflow schedule registrations. */
public interface ScheduleChangeListener {

    default void onScheduleRegistered(WorkflowSchedule schedule) {}

    default void onScheduleUpdated(WorkflowSchedule schedule) {}

    default void onScheduleDeleted(String name) {}

    default void onSchedulePaused(WorkflowSchedule schedule) {}

    default void onScheduleResumed(WorkflowSchedule schedule) {}
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/listener/ScheduleChangeListener.ts
import { WorkflowSchedule } from 'io/orkes/conductor/scheduler/model/WorkflowSchedule';

export interface ScheduleChangeListener {
    onScheduleRegistered?(schedule: WorkflowSchedule): void;

    onScheduleUpdated?(schedule: WorkflowSchedule): void;

    onScheduleDeleted?(name: string): void;

    onSchedulePaused?(schedule: WorkflowSchedule): void;

    onScheduleResumed?(schedule: WorkflowSchedule): void;
}
